src/vs/workbench/services/editor/common/customEditorLabelService.ts

```diff
@@ -148,29 +148,35 @@ export class CustomEditorLabelService extends Disposable implements ICustomEdito
 			}
 
 			if (pattern.parsedPattern(relevantPath)) {
-				return this.applyTempate(pattern.template, resource, relevantPath);
+				return this.applyTemplate(pattern.template, resource, relevantPath);
 			}
 		}
 
 		return undefined;
 	}
 
-	private readonly _parsedTemplateExpression = /\$\{(dirname|filename|extname|dirname\(([-+]?\d+)\))\}/g;
-	private applyTempate(template: string, resource: URI, relevantPath: string): string {
+	private readonly _parsedTemplateExpression = /\$\{(dirname|filename|extname\((?<extnameN>[-+]?\d+)\)|dirname\((?<dirnameN>[-+]?\d+)\))\}/g;
+	private applyTemplate(template: string, resource: URI, relevantPath: string): string {
 		let parsedPath: undefined | ParsedPath;
-		return template.replace(this._parsedTemplateExpression, (match: string, variable: string, arg: string) => {
+		return template.replace(this._parsedTemplateExpression, (match: string, variable: string, ...args: any[]) => {
 			parsedPath = parsedPath ?? parsePath(resource.path);
-			switch (variable) {
-				case 'filename':
-					return parsedPath.name;
-				case 'extname':
-					return parsedPath.ext.slice(1);
-				default: { // dirname and dirname(arg)
-					const n = variable === 'dirname' ? 0 : parseInt(arg);
-					const nthDir = this.getNthDirname(dirname(relevantPath), n);
-					if (nthDir) {
-						return nthDir;
-					}
+			// named group matches
+			const { dirnameN = '0', extnameN = '0' }: { dirnameN?: string; extnameN?: string } = args.pop();
+
+			if (variable === 'filename') {
+				return parsedPath.name;
+			} else if (variable === 'extname') {
+				return parsedPath.base.split('.').slice(1).join('.');
+			} else if (variable.startsWith('extname')) {
+				const n = parseInt(extnameN);
+				const extensionName = parsedPath.base;
+				const nthExtname = this.getNthExtname(extensionName, n);
+				return nthExtname ?? variable;
```

Comment:
if(nthExtname){
	return nthExtname;
}

Reply:
return match if no fragment could be found

Reply:
done